src/erlpmd_store.erl

```diff
@@ -0,0 +1,47 @@
+-module(erlpmd_store).
+
+-callback init(Args :: list(any()) ) -> {ok, State :: any()}
+                    | {error, Reason :: term()}.
+
+-callback register_node(
+            NodeName :: binary(),
+            {
+             PortNo :: non_neg_integer(), % TODO I'm sure there's an inet type() to use
+             NodeType :: 72 | 77,
+             Protocol :: non_neg_integer(),
+             HighestVersion :: non_neg_integer(),
+             LowestVersion :: non_neg_integer(),
+             Extra :: binary()
+            },
+            Fd :: inet:socket(), %% TODO Check this type I can never remember the correct thing
+            Creation :: non_neg_integer(),
+            State :: any()) ->
+    ok
+    | {ok, State1 :: any()}
+    | {error, registered}
+    | {error, Reason :: term()}.
+
+-callback node_port(NodeName :: binary(), State :: any()) ->
+    {ok, PortNo :: non_neg_integer()}
+    | {error, Reason :: term()}.
+
+-callback names(State :: any()) ->
+    {ok, Names :: list(binary())}
+    | {error, Reason :: term()}.
+
+%% TODO Maybe this should give back the structured data and let erlpmd take care of the structure->binary conv?
```

Comment:
Fix this `-callback` type, remove the TODO. This does return the structured data.